src/sage/combinat/root_system/cartan_matrix.py

```diff
@@ -1189,10 +1201,19 @@ def find_cartan_type_from_matrix(CM):
             T = DiGraph(ct.dynkin_diagram()) # We need a simple digraph here
             iso, match = T.is_isomorphic(S, certificate=True, edge_labels=True)
             if iso:
-                types.append(ct.relabel(match))
+                types.append(ct)
+                relabel.append(match)
                 found = True
                 break
         if not found:
             return None
 
-    return CartanType(types)
+    if len(types) == 1:
+        # Irreducible, so just relabel
+        return CartanType(types[0]).relabel(relabel[0])
+    ct = CartanType(types)
+    # ct._index_relabelling is a dict ``(ind, j): i``, where i is an index of
+    #   ``ct``, ``ind`` is the position in the list of types, and j is the
+    #   corresponding index of the type number ``ind``
```

Comment:
Does this mean that ``(ind, j): i`` means that the j-th node of types[ind] coresponds to the i-th node of ct? This would be a clearer way to state it, if true.

Reply:
Yes, that's correct. I added your phrasing as I find the way it is written more clear but it isn't a bad thign to have both.